### PR TITLE
Add maximum height & gemini scrollbar

### PR DESCRIPTION
**Pull request by jfurrow**
Also adds the ability to render the menu up or down, depending on which direction has the most available space.

### CODE REVIEW

**Comment by mesosphere-web** [comment]
Reviewer: @olegakbarov

File: src/Dropdown/Dropdown.js [hunk 1]
```diff
@@ -150,14 +176,18 @@ export default class Dropdown extends Util.mixin(BindMixin) {
   handleMenuToggle(e) {
     e.stopPropagation();
     let menuDirection = this.state.menuDirection;
+    let maxDropdownHeight = this.state.maxDropdownHeight;
 
     // If the menu isn't open, then we're about to open it and we need to
     // calculate the direction every time.
     if (!this.state.isOpen) {
-      menuDirection = this.getMenuDirection(this.state.menuHeight);
+      let menuStyle = this.getOptimalMenuStyle(this.state.menuHeight);
```
Comment (kennyt) [inline src/Dropdown/Dropdown.js]: you can do:

```js
let {direction, height} = this.getOptimalMenuStyle(this.state.menuHeight);
```

**Comment by kennyt** [comment]
is there an example in the docs that utilizes the scrollbar?

File: src/Dropdown/Dropdown.js [hunk 2]
```diff
@@ -35,16 +37,21 @@ export default class Dropdown extends Util.mixin(BindMixin) {
       let dropdownMenuConcealer = this.refs.dropdownMenuConcealer.getDOMNode();
       let menuDirection = this.state.menuDirection;
```
Comment (kennyt) [inline src/Dropdown/Dropdown.js]: alright, right here you can do:

```js
let {menuDirection, menuHeight, maxDropdownHeight} = this.state;
```

:)

Comment (jfurrow) [inline src/Dropdown/Dropdown.js]: Done!

**Comment by jfurrow** [comment]
@kennyt I added an example with a large dropdown that utilizes the scrollbar. Keep in mind we have `overflow: hidden` on the code containers, so the dropdown will be clipped if your veiwport is large enough.

**Comment by jfurrow** [comment]
@mlunoe @rcorral friendly bump! :)

**Pull request merged**